### PR TITLE
youtube shortcode refactor

### PR DESCRIPTION
**Pull request by gumaerc**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-to-hugo/issues/210
Closes https://github.com/mitodl/ocw-to-hugo/issues/207

#### What's this PR do?
This PR refactors the way `youtube` shortcodes are generated by instead replacing the Youtube embed codes with a div with a special class.  This class of div is then caught in a turndown rule and replaced with the proper shortcode.  The `media_location` property is deposited into the div as the content and is pulled out in the turndown rule with `node.textContent` so it is not run through markdown transformation.

Note that some other general refactoring is being done in this PR.  All test data has been updated with the latest parsed json, some unused imports were removed and there was a failing test after updating the test json that was corrected.

#### How should this be manually tested?
Spin up a local test site using `ocw-course-hugo-theme` (read the readme if you've never done this before) for `8-01sc-classical-mechanics-fall-2016` and visit http://localhost:1313/sections/week-1-kinematics/3.1-coordinate-system-and-position-vector-in-2d/.  The YouTube video should now properly render.

#### Screenshots (if appropriate)
image

### CODE REVIEW

File: src/turndown.js [hunk 1]
```diff
@@ -358,6 +353,24 @@ turndownService.addRule("h4", {
...
+  filter: (node, options) => {
+    const nodeClass = node.getAttribute("class")
+    if (
+      node.nodeName === "DIV" &&
+      nodeClass &&
+      nodeClass.includes(YOUTUBE_SHORTCODE_PLACEHOLDER_CLASS)
+    ) {
+      return true
+    }
+    return false
```
Comment (aliceriot) [inline src/turndown.js:366]: wouldn't this be equivalent to 

```js
return (
  node.nodeName === "DIV" && nodeClass === YOUTUBE_SHORTCODE_PLACEHOLDER_CLASS
)
```

just slightly more compact that way I suppose

**Review by aliceriot — CHANGES_REQUESTED** [review]
just one small question

**Comment by gumaerc** [comment]
@aliceriot Should be all set.

**Review by aliceriot — APPROVED** [review]
LGTM!

**Pull request merged**